zuul-core/src/test/java/com/netflix/netty/common/proxyprotocol/ElbProxyProtocolChannelHandlerTest.java

```diff
@@ -14,59 +14,103 @@
+        assertNull(channel.pipeline().context("HAProxyMessageChannelHandler"));
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_VERSION).get());
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_MESSAGE).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDRESS).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_PORT).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_ADDRESS).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_PORT).get());
         assertNull(channel.attr(SourceAddressChannelHandler.ATTR_REMOTE_ADDR).get());
+    }
 
+    @Test
+    public void passThrough_ProxyProtocolEnabled_nonProxyBytes() {
+        EmbeddedChannel channel = new EmbeddedChannel();
+        channel.pipeline().addLast(ElbProxyProtocolChannelHandler.NAME,
+                new ElbProxyProtocolChannelHandler(/* withProxyProtocol= */ true));
+        ByteBuf buf = Unpooled.wrappedBuffer(
+                "TCP4 192.168.0.1 124.123.111.111 10008 443\r\n".getBytes(StandardCharsets.US_ASCII));
+        channel.writeInbound(buf);
+
+        Object dropped = channel.readInbound();
+        assertEquals(dropped, buf);
+
+        assertNull(channel.pipeline().context(ElbProxyProtocolChannelHandler.NAME));
+        assertNull(channel.pipeline().context("HAProxyMessageChannelHandler"));
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_VERSION).get());
+        assertNull(channel.attr(HAProxyMessageChannelHandler.ATTR_HAPROXY_MESSAGE).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDRESS).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_PORT).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_ADDRESS).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_PORT).get());
+        assertNull(channel.attr(SourceAddressChannelHandler.ATTR_REMOTE_ADDR).get());
+    }
+
+    @Ignore
+    @Test
+    public void detectsSplitPpv1Message() {
+        EmbeddedChannel channel = new EmbeddedChannel();
+        channel.pipeline().addLast(ElbProxyProtocolChannelHandler.NAME,
+                new ElbProxyProtocolChannelHandler(/* withProxyProtocol= */ true));
+        ByteBuf buf1 = Unpooled.wrappedBuffer(
+                "PROXY TCP4".getBytes(StandardCharsets.US_ASCII));
+        channel.writeInbound(buf1);
+        ByteBuf buf2 = Unpooled.wrappedBuffer(
+                "192.168.0.1 124.123.111.111 10008 443\r\n".getBytes(StandardCharsets.US_ASCII));
+        channel.writeInbound(buf2);
+
+        Object msg = channel.readInbound();
+        assertTrue(msg instanceof HAProxyMessage);
```

Comment:
Netty objects are ref counted, and you need to release them at the  end of the test